client/fs_endpoint.go

```diff
@@ -637,10 +639,10 @@ func (f *FileSystem) logsImpl(ctx context.Context, follow, plain bool, offset in
 
 // streamFile is the internal method to stream the content of a file. If limit
 // is greater than zero, the stream will end once that many bytes have been
-// read. eofCancelCh is used to cancel the stream if triggered while at EOF. If
+// read. eofCancelOnNextEofCh, if triggered while at EOF, is used to trigger one more read and cancel the stream on reaching next EOF. If
 // the connection is broken an EPIPE error is returned
 func (f *FileSystem) streamFile(ctx context.Context, offset int64, path string, limit int64,
-	fs allocdir.AllocDirFS, framer *sframer.StreamFramer, eofCancelCh chan error) error {
+	fs allocdir.AllocDirFS, framer *sframer.StreamFramer, eofCancelCh chan error, eofCancelOnNextEofCh chan error) error {
```

Comment:
Naming the parameter `eofCancelOnNextEofCh` here when the channel that we construct to pass as that parameter is called `newFileChannelCh` makes this kind of hard to follow. Particularly because we have a similarly named channel right next to it and in the caller.

Reply:
Sure. I renamed the variable to match the naming